cpp/src/arrow/dataset/file_parquet_test.cc

```diff
@@ -88,5 +94,102 @@ TEST_F(TestParquetFileFormat, ScanRecordBatchReader) {
   ASSERT_EQ(row_count, kNumRows);
 }
 
+class TestFileSystemBasedDataSource : public FileSourceFixtureMixin {
+ public:
+  void SetUp() override {
+    format_ = std::make_shared<ParquetFileFormat>();
+
+    ASSERT_OK(TemporaryDir::Make("test-fsdatasource-", &temp_dir_));
```

Comment:
If you want you could also use a mock filesystem (see `arrow/filesystem/mockfs.h`). It's in-memory.

Reply:
That will be handy